File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/TopArtistsInteractor.java
package com.example.artist.ui.topartistslisting;

import com.example.artist.models.TopArtistsResponse;
import com.example.artist.models.TopArtistsResponseResult;

import io.reactivex.Single;


public interface TopArtistsInteractor {
    Single<TopArtistsResponseResult> getTopArtists(String userName, String apiKey);
}

File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/TopArtistsInteractor.kt
package com.example.artist.ui.topartistslisting

import com.example.artist.models.TopArtistsResponseResult
import io.reactivex.Single

interface TopArtistsInteractor {
    fun getTopArtists(userName: String, apiKey: String): Single<TopArtistsResponseResult>
}